2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,560 @@
+
+- After logging in, every Trainee can see the progress of Checkout Process, as described in Milestone 1.
+
+## Schema Changes
+
+```python
+class SWCOrDCField(models.CharField):
+    max_length = STR_MED
+    choices = (
+        ('swc', 'Software-Carpentry'),
+        ('dc', 'Data-Carpentry'),
+    )
+    blank = False
+    default = 'swc'
+
+class Person(AbstractBaseUser, PermissionsMixin):
+    # determines whether the person has access to AMY interface for admins and trainers
+    is_admin = models.BooleanField(default=True)  
+
+    # username -- no longer required
+    # email -- required
+
+    swc_instructor_training = models.ForeignKey('InstructorTraining', null=True)
+    dc_instructor_training = models.ForeignKey('InstructorTraining', null=True)
+    swc_core_lesson = models.ForeignKey('CoreLesson', null=True, related_name='trainees')
+    dc_core_lesson = models.ForeignKey('CoreLesson', null=True, related_name='trainees')
+    cannot_make_discussion_session = models.BooleanField(default=False)
+    cannot_make_checkout_session = models.BooleanField(default=False)
+    availability_for_discussion_session = models.TextField(default="", blank=True)
+    availability_for_checkout_session = models.TextField(default="", blank=True)
+    link_to_swc_lesson_change = models.URLField(default="", blank=True)
+    link_to_dc_lesson_change = models.URLField(default="", blank=True)
+    is_swc_lesson_change_good_enough = models.BooleanField(default=True)
+    is_dc_lesson_change_good_enough = models.BooleanField(default=True)
+    is_certified_instructor = models.BooleanField(default=False)
+    biography = models.TextField(default="", blank=True)
+    photo = models.FileField()
+
+# The source of truth for Discussion Sessions and Checkout Sessions is Amy
+# database. However, we can later switch to get them from Google Calendar. In
+# that case, we need to periodically synchronize Amy database with the
+# calendar.
+class Session(models.Model):
+    leader = models.ForeignKey('Person')
+    start = models.DateTimeField()
+    swc_or_dc = SWCOrDCField()
+    status = models.CharField(
+        max_length=STR_MED,
+        choices=(
+            ('not-started', 'Did not started yet'),
+            ('canceled', 'Canceled'),
+            ('took-place', 'Took place')
+        ),
+        blank=False, default='not-started',
+    )
+    notes = models.TextField(default="", blank=True)
+    trainees = models.ManyToManyField(Person, 
+        through='SessionAttendance', 
+        through_fields=('session', 'person'),
+        related_name='sessions')
+
+class SessionAttendance(models.Model):
+    person = models.ForeignKey('Person')
+    session = models.ForeignKey('Session')
+    status = models.CharField(
+        max_length=STR_MED,
+        choices=(
+            ('na', 'NA'),
+            ('absent', 'Absent'),
```

Comment:
Is the difference between `absent` and no entry == "signed up but didn't attend"?